storage/database/src/main/java/de/danoeh/antennapod/storage/database/DBWriter.java

```diff
@@ -521,6 +521,73 @@ private static void removeQueueItemSynchronous(final Context context,
         }
     }
 
+    public static Future<?> moveQueueItemsToTop(final Context context,
+                                            final long... itemIds) {
+        return runOnDbThread(() -> moveQueueItemsSynchronous(context, true, itemIds));
+    }
+
+    public static Future<?> moveQueueItemsToBottom(final Context context,
+                                                final long... itemIds) {
+        return runOnDbThread(() -> moveQueueItemsSynchronous(context, false, itemIds));
+    }
+
+    private static void moveQueueItemsSynchronous(final Context context,
+                                                  final boolean moveToTop,
+                                                  final long... itemIds) {
+
+        String calledBy = moveToTop ? "moveQueueItemsToTop" : "moveQueueItemsToBottom";
+
+        if (itemIds.length < 1) {
+            return;
+        }
+
+        final PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        final List<FeedItem> queue = DBReader.getQueue();
+
+        if (moveToTop) {
+            for (int i = 0; i < itemIds.length / 2; i++) {
+                long temp = itemIds[i];
+                itemIds[i] = itemIds[itemIds.length - 1 - i];
+                itemIds[itemIds.length - 1 - i] = temp;
+            }
+        }
+
+        boolean queueModified = false;
+        List<QueueEvent> events = new ArrayList<>();
+        for (long itemId : itemIds) {
+            int index = indexInItemList(queue, itemId);
+            if (index >= 0) {
+                final FeedItem item = DBReader.getFeedItem(itemId);
```

Comment:
When calling this method, you convert the list of items to a list of IDs. Here you have to load the items again. If you pass the items instead, you don't need the potentially expensive calls to the database here.